targets/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_SAML21J18A/device/TOOLCHAIN_GCC_ARM/startup_saml21.c

```diff
@@ -258,14 +267,17 @@ void Reset_Handler(void)
     pSrc = (uint32_t *) & _sfixed;
     SCB->VTOR = ((uint32_t) pSrc & SCB_VTOR_TBLOFF_Msk);
 
-    /* Initialize the C library */
-    __libc_init_array();
-
     /* Overwriting the default value of the NVMCTRL.CTRLB.MANW bit (errata reference 13134) */
     NVMCTRL->CTRLB.bit.MANW = 1;
 
-    /* Branch to main function */
+    SystemInit();
+
+#ifdef MBED_CONF_RTOS_PRESENT
+    software_init_hook();
+#else
+    __libc_init_array();
```

Comment:
Doesn't mbed_boot code file already provide these for Mbed OS 5 ? See mbed_boot.c file, `mbed_start` function.

See the comments in the file:

> Reset vector is
 * responsible for low level platform init and then calling in libc (__main). Depending on compiler and version of C
 * library, predefined function will be called which is implemented by mbed OS.

Reply:
@0xc0170 How do I set the reset vector to behave like this then? Because I copied this behavior from some NXP targets ([example](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_NXP/TARGET_LPC11U6X/device/TOOLCHAIN_GCC_ARM/TARGET_LPC11U68/startup_LPC11U68.cpp#L152)) and it seems to properly boot in both RTOS and non-RTOS mode. I don't really get how other targets do it because it's all in .S files not in C.

Reply:
Understood, looks fine